Create Field Tables for Events (#1051)
temporalio/documentation: docs/concepts/what-is-an-event.md

> 8---

Reviewer: ```suggestion
---
```

  - explanation
---

Events are created by the Temporal Cluster in response to external occurrences and Commands generated by a Workflow Execution. Each Event corresponds to an `enum` that is defined in the [Server API](https://github.com/temporalio/api/blob/master/temporal/api/enums/v1/event_type.proto).

All Events are recorded in the [Event History](/docs/concepts/what-is-an-event-history).

A list of all possible Events that could appear in a Workflow Execution Event History is provided below.

### WorkflowExecutionStarted

This is always the first Event in a Workflow Execution History. It indicates that the execution was invoked.

| Field | Type | Definition |
| ----- | ---- | ---------- |

### WorkflowExecutionCompleted

This indicates that the Workflow Execution has successfully completed. The Event contains Workflow Execution results.

> | Field                            | Type     | Definition                                                                        |

Reviewer 2: I am thinking that we should go with Description column name.

Definition -> Description

| -------------------------------- | -------- | --------------------------------------------------------------------------------- |

> | result                           | Payloads | Serialized result of completed Workflow.                                          |

Reviewer 2: OK, so I know this is annoying to change, but I think we should drop the Type column altogether.

I think we should link to the source file in Github for more information.
But since each language converts these protobuf definitions into a language specific type - its kind of too much - and maybe confusing.

Most people will interact with Event History fields via JSON or the WebUI.

> | workflow_task_completed_event_id | int64    | The Id of the 'WORKFLOW_TASK_COMPLETED' that the event command was reported with. |

Reviewer 2: ```suggestion
| workflow_task_completed_event_id | int64    | The Id of the [WorkflowTaskCompleted](#workflowtaskcompleted) that the event command was reported with. |
```
1. If the description refers to another Event by name - we can link to it locally on the page (example in suggestion)

2. Let's capitalize core terms - such as Event.

This particular sentence however is confusing 
> The Id of the 'WORKFLOW_TASK_COMPLETED' that the event command was reported with.

What is an "event command" ?
We have Events
We have Commands
Let's try to figure out which one is meant by this.

> | new_execution_run_id             | string   | The Id of the new workflow started by cron.                                       |

Reviewer 2: ```suggestion
| new_execution_run_id             | string   | The Run Id of the next Workflow Execution in the Workflow Execution Chain (often a result of a Cron Schedule)                        |
```
Just a suggestion for clarification / accuracy.

### WorkflowExecutionFailed

> This event indicates that the Workflow execution has unsuccessfully completed and contains the Workflow execution error.

Reviewer 2: ```suggestion
This Event indicates that the Workflow Execution has unsuccessfully completed and contains the error returned by the latest Workflow Function Execution.
```

I think that is correct, but maybe get double checked by @dtornow

| Field                            | Type       | Definition                                                                        |
| -------------------------------- | ---------- | --------------------------------------------------------------------------------- |
| failure                          | Failure    | Serialized result of a Workflow failure, such as an exception or error.           |
| retry_state                      | RetryState |                                                                                   |

> | workflow_task_completed_event_id | int64      | The Id of the 'WORKFLOW_TASK_COMPLETED' that the event command was reported with. |

Reviewer: ```suggestion
| workflow_task_completed_event_id | int64      | The Id of the `WORKFLOW_TASK_COMPLETED` that the event command was reported with. |
```

Add back ticks when referencing code not apostrophes, so the code is formatted.

Reviewer 2: In this case though, WORKFLOW_TASK_COMPLETED is actually the Event WorkflowTaskCompleted - and we want to link to them I think

| Field Name  | Type     | Definition |
| ----------- | -------- | ---------- |
| signal_name | string   | definition |
| input       | Payloads | definition |
| identity    | string   | definition |
| header      | Header   | definition |

### WorkflowExecutionTerminated

This event type indicates that the Workflow execution has been forcefully terminated and that likely the terminate Workflow API was called.

| Field Name | Type     | Definition |
| ---------- | -------- | ---------- |
| reason     | string   | definition |
| details    | Payloads | definition |
| identity   | string   | definition |

### WorkflowExecutionContinuedAsNew

This event type indicates that the Workflow has successfully completed, and a new Workflow has been started within the same transaction.
This event type contains last Workflow execution results as well as new Workflow execution inputs.

| Field Name                       | Type                   | Definition                                                                        |
| -------------------------------- | ---------------------- | --------------------------------------------------------------------------------- | --- |

> | new_execution_run_id             | string                 | The run ID of the new workflow started by this event.                             |

Reviewer: I believe the term `Id` isn't fully capitalized. It should just be `Id` when referenced in prose, as seen from the [Concepts page](https://docs.temporal.io/docs/concepts/what-is-an-activity-id).

| prev_auto_reset_points                                                               |                                                                                                                                                   |
| header                                                                               | Information passed by the sender of the [Signal](/docs/concepts/what-is-a-signal.md) that is copied into the Workflow Task.                       |

### WorkflowExecutionCompleted

This indicates that the Workflow Execution has successfully completed. The Event contains Workflow Execution results.

| Field                            | Description                                                                                         |
| -------------------------------- | --------------------------------------------------------------------------------------------------- |
| result                           | Serialized result of completed Workflow.                                                            |

> | workflow_task_completed_event_id | The Id of the [`WORKFLOW_TASK_COMPLETED`](#workflowtaskcompleted) that the Event was reported with. |

Reviewer 2: ```suggestion
| workflow_task_completed_event_id | The Id of the [WorkflowTaskCompleted](#workflowtaskcompleted) that the Event was reported with. |
```

| ---------------------------------- | ---------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------- |
| workflow_type                      | WorkflowType           | The name/type of Workflow that was initiated.                                                                                                   |
| parent_workflow_namespace          | string                 | The namespace of the parent Workflow, if applicable.                                                                                            |
| parent_workflow_execution          | WorkflowExecution      | Identifies the parent Workflow and the execution run.                                                                                           |
| parent_initiated_event_id          | int64                  | Id of the 'START_WORKFLOW_EXECUTION_INITIATED' Event this Event corresponds to.                                                                 |

> | task_queue                         | TaskQueue              | The Task Queue that this Workflow Task was enqueued in.                                                                                         |

Reviewer 2: ```suggestion
| task_queue                         | TaskQueue              | The [Task Queue](/docs/concepts/what-is-a-task-queue) that this Workflow Task was enqueued in.                                                                                         |
```

| identity    | string   | Identifies the Worker that signaled to the Workflow.                                       |
| header      | Header   | Information passed by the sender of the Signal that is copied into the Workflow Task.      |

### WorkflowExecutionTerminated

This event type indicates that the Workflow execution has been forcefully terminated and that likely the terminate Workflow API was called.

| Field    | Type     | Definition                                                           |
| -------- | -------- | -------------------------------------------------------------------- |
| reason   | string   | Information provided by the user or client for Workflow termination. |
| details  | Payloads | Additional information reported by the Workflow upon termination.    |
| identity | string   | Identifies the Worker that requested termination.                    |

### WorkflowExecutionContinuedAsNew

This event type indicates that the Workflow has successfully completed, and a new Workflow has been started within the same transaction.
This event type contains last Workflow execution results as well as new Workflow execution inputs.

> <<<<<<< HEAD

Reviewer 2: ```suggestion
```

| retry_state        | RetryState | definition |

### ActivityTaskCancelRequested

This event type indicates that a request to cancel the Activity has occurred.
The SDK client will be able to confirm cancellation of an Activity during an Activity heartbeat.

| Field Name                       | Type  | Definition |
| -------------------------------- | ----- | ---------- |
| scheduled_event_id               | int64 | definition |
| workflow_task_completed_event_id | int64 | definition |

### ActivityTaskCanceled

This event type indicates that the Activity has been canceled.

| Field Name                       | Type     | Definition                                                                                      |
| -------------------------------- | -------- | ----------------------------------------------------------------------------------------------- |
| details                          | Payloads | Additional information reported by the Activity upon confirming cancellation.                   |
| latest_cancel_requested_event_id | int64    | ID of the most recent 'ACTIVITY_TASK_CANCEL_REQUESTED' event which refers to the same Activity. |

> | scheduled_event_id               | int64    | the ID of the 'ACTIVITY_TASK_SCHEDULED' event this cancel confirmation corresponds to.          |

Reviewer: `cancellation` should be spelled `cancelation` in prose. The two `l`'s is British English and the one singualr `l`, is American English. [Google's style guide](https://developers.google.com/style/spelling)

| started_event_id                 | int64    | definition                                                                                      |
| identity                         | string   | definition                                                                                      |

### TimerStarted

This event type indicates a timer has started.

| Field Name                       | Type     | Definition                                       |
| -------------------------------- | -------- | ------------------------------------------------ |
| timer_id                         | string   | The Worker or User ID assigned for the timer.    |
| start_to_fire_timeout            | Duration | Amount of time to elapse before the timer fires. |
| workflow_task_completed_event_id | int64    | definition                                       |

### TimerFired

This event type indicates a timer has fired.

| Field Name       | Type   | Definition |
| ---------------- | ------ | ---------- |
| timer_id         | string | definition |
| started_event_id | int64  | definition |

### TimerCanceled

This event type indicates a Timer has been canceled.

> | Field Name                       | Type       | Definition |

Reviewer: `Field Name` should be `Field name` due to [sentence casing](https://developers.google.com/style/capitalization#capitalization-in-references-to-titles-and-headings)

Reviewer 2: Drop "name" altogether in favor of just "Field"
